### PR TITLE
Migration/stavehairpin

### PR DESCRIPTION
**Pull request by rvilarl**
StaveHairpin!

### CODE REVIEW

**Comment by rvilarl** [comment]
@0xfe wow a lot of progress! Did you see this little one?

File: src/stavehairpin.ts [hunk 1]
```diff
@@ -9,14 +9,39 @@
 import { Vex } from './vex';
 import { Element } from './element';
 import { Modifier } from './modifier';
+import { Note } from './note';
+import { RenderContext } from './types/common';
+
+export interface StaveHairpinRenderParams {
```
Comment (0xfe) [inline src/stavehairpin.ts:15]: This can be inline.

File: src/stavehairpin.ts [hunk 2]
```diff
@@ -165,23 +203,23 @@ export class StaveHairpin extends Element {
...
     const start = firstNote.getModifierStartXY(this.position, 0);
     const end = lastNote.getModifierStartXY(this.position, 0);
 
     this.renderHairpin({
       first_x: start.x,
       last_x: end.x,
-      first_y: firstNote.getStave().y + firstNote.getStave().height,
-      last_y: lastNote.getStave().y + lastNote.getStave().height,
-      staff_height: firstNote.getStave().height,
+      first_y: firstNote.checkStave().getY() + (firstNote.checkStave().getHeight() ?? 0),
```
Comment (0xfe) [inline src/stavehairpin.ts:220]: This `?? 0` here is a silent failure. It seems like `stave.height` should not be optional, all staves must have a height. Can we make it (and the result of `getHeight`) non-optional?

File: src/stavehairpin.ts (hunk 1 above)
Comment (rvilarl) [reply src/stavehairpin.ts:15]: done

File: src/stavehairpin.ts (hunk 2 above)
Comment (rvilarl) [reply src/stavehairpin.ts:220]: yes it is always initialised in constructor, done

**Comment by rvilarl** [comment]
Ready to merge

**Comment by 0xfe** [comment]
LGTM!

**Pull request merged**